### ISSUE TITLE
Firefox Android seems to create a "ghost entry" in your history for every domain bookmarked.

### ISSUE THREAD

**Issue by Aurionin** [mozilla-mobile/fenix#28084]
### Steps to reproduce

1. Bookmark a page.
2. Website will appear in your suggested URLs.
3. Remove the bookmark.
4. Website will remain in your suggested URLs, even if no browser history or bookmark of it exists.

### Expected behaviour

Should no longer suggest the website if no history or bookmark of it exists.

### Actual behaviour

When I bookmark on website on Android Firefox, that website is now suggested as a URL autofill, which seems normal. The only problem is that, even if I delete the bookmark it remains in my auto fill suggestions. This also continues even if I have no listings for it in my browser history (if I had deleted that website for my history, or if the bookmark was done in private browsing meaning there never was a history in the first place).

Firefox for Android has several settings to let you toggle what kinds of things are URL suggestions, and playing with them a little bit it looks like the only one that gets rid of this ghost entry is if I turn off "search browsing history" under "address bar." So: 

 • The only thing saved is the domain of the bookmark, not the full address. (ie. Bookmarking "http://www reddit.com/r/Firefox" will only leave a ghost entry for reddit's homepage)

 • It appears this ghost entry is in the history somewhere. 

 • It doesn't show up in the normal history if you scroll down through it, but it does appear if you use the search bar in your history.

 • The only way it looks like to get rid of it is to delete all of your browser history. Even using the option to delete everything from the last hour, or the last 2 days doesn't get rid of it.

tl;dr: The Android version of Firefox seems to save a record of every domain you've ever bookmarked, and the only way to clear this record is to delete your entire browsing history. 

### Device name

Google pixel 5

### Android version

Android 13

### Firefox release type

Firefox

### Firefox version

107.2

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22100)

**Comment by jmora13** [comment]
Is this issue open? I'd like to work on it!

**Comment by Aurionin** [comment]
Still doing it for me, and I haven't gotten any other replies yet!

**Comment by jmora13** [comment]
Sure! I'll get started on it.

**Comment by LendrickKamar** [comment]
The same problem is also shown on Firefox Beta 109.0b4!

**Comment by boek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813898

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
